controllers/machine_controller_phases.go

```diff
@@ -86,23 +86,34 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
 		return nil, err
 	}
 
-	objPatch := client.MergeFrom(obj.DeepCopy())
+	// Initialize the patch helper.
+	patchHelper, err := patch.NewHelper(obj, r.Client)
+	if err != nil {
+		return nil, err
+	}
 
 	// Set external object OwnerReference to the Machine.
-	machineOwnerRef := metav1.OwnerReference{
+	ownerRef := metav1.OwnerReference{
 		APIVersion: clusterv1.GroupVersion.String(),
 		Kind:       "Machine",
 		Name:       m.Name,
 		UID:        m.UID,
 	}
 
-	if !util.HasOwnerRef(obj.GetOwnerReferences(), machineOwnerRef) {
-		obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), machineOwnerRef))
-		if err := r.Client.Patch(ctx, obj, objPatch); err != nil {
-			return nil, errors.Wrapf(err,
-				"failed to set OwnerReference on %v %q for Machine %q in namespace %q",
-				obj.GroupVersionKind(), ref.Name, m.Name, m.Namespace)
-		}
+	// Add ownerRef to object.
+	obj.SetOwnerReferences(util.EnsureOwnerRef(obj.GetOwnerReferences(), ownerRef))
```

Comment:
Wouldn't this potentially create an ownerRef for a Machine owned by a MachineSet? Are we intending to change that behavior for v1alpha3?